sstables/sstable_set.cc

```diff
@@ -464,6 +470,104 @@ class time_series_sstable_set : public sstable_set_impl {
     }
 };
 
+// Reader and owning pointer to the sstable being read.
+// Prevents the sstable from being closed while the reader is being used.
+struct backed_reader {
+    shared_sstable sst;
+    flat_mutation_reader reader;
+};
+
+// Queue of readers of sstables in a time_series_sstable_set,
+// returning readers in order of the sstables' min_position()s.
+//
+// Skips sstables that don't pass the supplied filter.
```

Comment:
Why not pass in the sstables pre-filtered instead?

Reply:
I didn't want to copy the set of sstables just to perform the query. Now there's a single copy living inside `time_series_sstable_set` and I'm just iterating over it.

But dunno, maybe it complicates things unnecessarily. Can we afford the copy on every query?

Reply:
Time series queries usually span many rows, so we can afford to do a little extra work. But: it's better not to do the extra work.

What's the filter? The bloom filter? It will usually return true for TWCS, so pre-filtering will not help much.

Reply:
It includes the ck filtering as well.